sched/semaphore/sem_wait.c

```diff
@@ -72,10 +72,17 @@ int nxsem_wait(FAR sem_t *sem)
 {
   FAR struct tcb_s *rtcb = this_task();
   irqstate_t flags;
-  int ret = -EINVAL;
+  int ret;
 
   /* This API should not be called from interrupt handlers & idleloop */
 
+#ifdef CONFIG_DEBUG_FEATURES
+  if (sem == NULL)
+    {
+      return -EINVAL;
+    }
+#endif
```

Comment:
https://pubs.opengroup.org/onlinepubs/7908799/xsh/sem_wait.html states:
```
[EINVAL]
    The sem argument does not refer to a valid semaphore. 
```
So that is not a debug feature